### PR TITLE
drupal-check addon 

### PR DESCRIPTION
**Pull request by ghost**
In preparation for the eventual upgrade from Drupal 8 to Drupal 9 [Matt Glaman (mglaman)](https://www.drupal.org/u/mglaman) has created a [drupal-check script](https://github.com/mglaman/drupal-check) built on PHPStan to help developers and module maintainers prepare their code. Dries has also talked about this in his post [How to prepare for Drupal 9](https://dri.es/how-to-prepare-for-drupal-9)

Matt Glaman' drupal-check script analyzes modules on a site searching for any deprecated code that would otherwise cause a site to fail after upgrading.

This addon puts a wrapper around the drupal-check script allowing anyone who is currently using docksal to begin preparing their site for Drupal 9 quickly.

### CODE REVIEW

**Comment by ghost** [comment]
I would've created the [PR against a develop branch](https://github.com/docksal/docksal/blob/develop/CONTRIBUTING.md#contribute-code), but I didn't see one for the addons.

**Comment by achekulaev** [comment]
Not sure why you created draft PR

**Comment by ghost** [comment]
I can change it to a normal PR, I was just gauging  interest for the add on.

File: drupal-check/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Drupal-check Docksal Addon
+
+## Installation
+
+Download or clone the drupal-check addon and place it into your .docksal/addons directory.
```
Comment (achekulaev) [inline drupal-check/README.md:5]: Once addon is in this repo there will be no need to clone it to install

**Comment by achekulaev** [comment]
The repo is pretty open to propositions. Any addon that is not harmful and makes sense will be approved.

File: drupal-check/README.md (hunk 1 above)
Comment (ghost) [reply drupal-check/README.md:5]: Ok good call! We're using this module internally and each project needs to pull it down piecemeal, hence the README. I went ahead and deleted it from this PR, most of the information is duplicated in the actual scripts help comments.

**Comment by acondura** [comment]
I am definitely interested in this!

**Comment by lpeabody** [comment]
Possibility of getting this in? Would be a handy tool for our projects.

File: drupal-check/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Drupal-check Docksal Addon
+
+## Usage
+
+This tool works on all Drupal code.
```
Comment (achekulaev) [inline drupal-check/README.md]:
```suggestion
Check Drupal code for deprecations and discover bugs via static analysis using [mglaman/drupal-check](https://github.com/mglaman/drupal-check)
```

File: drupal-check/drupal-check [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+# Default variables.
+SCRIPTS_DIR="/var/www/.docksal/addons/drupal-check/scripts"
+MODULE_TYPE="contrib"
+CHECK_OPTS=()
+PROFILE_DIR=""
+SCRIPT_ARG=""
+
+# Import local settings
+[[ -f local.drupal-check ]] && source local.drupal-check || CUSTOM=false
```
Comment (achekulaev) [inline drupal-check/drupal-check:30]: Is it by design that `local.drupal-check` file should always be in a current working directory?

File: drupal-check/drupal-check.pre-install [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+mkdir -p ${PROJECT_ROOT}/.docksal/addons/drupal-check/scripts
```
Comment (achekulaev) [inline drupal-check/drupal-check.pre-install:9]:
```suggestion
# Check that wget is installed as it is required to download the script
if ! (which wget >/dev/null); then
    echo "ERROR: wget is required to install drupal-check addon"
    exit 1
fi

mkdir -p ${PROJECT_ROOT}/.docksal/addons/drupal-check/scripts
```

File: drupal-check/example.local.drupal-check [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+## This is where your custom path goes.
+## Variables:
+## DOCROOT - The docroot of your project. In most cases this will be 'docroot'.
+## PROFILE_DIR - The directory of your profile.
+## MODULE_NAME - The name of the module to be evaluated.
+## The custom path MUST match the format of your profile module path or you will get errors.
+
+CUSTOM="/var/www/${DOCROOT}/profiles/custom/${PROFILE_DIR}/modules/${MODULE_NAME}"
```
Comment (achekulaev) [inline drupal-check/example.local.drupal-check]: `CUSTOM` is a very prone to errors name for environment variable. I would recommend changing its name to `DRUPAL_CHECK_MODULE_PATH` and updating scripts and documentation accordingly

**Comment by achekulaev** [comment]
@cbarrettgenuine ok got to reviewing it, sorry for the delay. Please see comments and propositions above. 

In addition to the above, please add this addon to the list of addons in global README file.

**Comment by achekulaev** [comment]
@cbarrettgenuine ready when you are!

**Comment by lmakarov** [comment]
Note: `drupal-check` binary will ship by default in `docksal/cli` starting with v2.9.0
https://github.com/docksal/service-cli/pull/139

File: drupal-check/drupal-check (hunk 3 above)
Comment (JDDoesDev) [reply drupal-check/drupal-check:30]: This is a check to see if local settings are there. We've included an `example.local.drupal-check` file as an example.

**Comment by JDDoesDev** [comment]
> Note: `drupal-check` binary will ship by default in `docksal/cli` starting with v2.9.0
> [docksal/service-cli#139](https://github.com/docksal/service-cli/pull/139)

I'll remove the call to install the binary and update the code accordingly

**Comment by achekulaev** [comment]
seems stale

**Pull request closed without merging**